File: TransCoder_tokenized_test_set_functions/python/COUNT_DISTINCT_SUBSEQUENCES.py
def countSub ( ss ) :
    last = [ - 1 for i in range ( MAX_CHAR + 1 ) ]
    n = len ( ss )
    dp = [ - 2 for i in range ( n + 1 ) ]
    dp [ 0 ] = 1
    for i in range ( 1, n + 1 ) :
        dp [ i ] = 2 * dp [ i - 1 ]
        if last [ ord ( ss [ i - 1 ] ) ] != - 1 :
            dp [ i ] = dp [ i ] - dp [ last [ ord ( ss [ i - 1 ] ) ] ]
        last [ ord ( ss [ i - 1 ] ) ] = i - 1
    return dp [ n ]

File: results/java_codex/COUNT_DISTINCT_SUBSEQUENCES.java
int countSub(string str) {
        int last[MAX_CHAR];
        fill(last, last + MAX_CHAR, -1);
        int n = str.length();
        int dp[n + 1];
        dp[0] = 1;
        for (int i = 1; i <= n; i++) {
            dp[i] = 2 * dp[i - 1];
            if (last[str[i - 1]] != -1)
                dp[i] = dp[i] - dp[last[str[i - 1]]];
            last[str[i - 1]] = (i - 1);
        }
        return dp[n];
    }